yarn-project/simulator/src/public/executor.ts

```diff
@@ -111,37 +77,54 @@ export class PublicExecutor {
       } satisfies AvmSimulationStats,
     );
 
-    const publicExecutionResult = trace.toPublicExecutionResult(
-      avmExecutionEnv,
-      /*startGasLeft=*/ allocatedGas,
+    const publicExecutionResult = stateManager.trace.toPublicEnqueuedCallExecutionResult(
       /*endGasLeft=*/ Gas.from(avmContext.machineState.gasLeft),
-      bytecode,
-      avmResult,
-      fnName,
+      avmCallResult,
     );
 
-    if (publicExecutionResult.reverted) {
+    if (avmCallResult.reverted) {
       this.metrics.recordFunctionSimulationFailure();
     } else {
       this.metrics.recordFunctionSimulation(bytecode.length, timer.ms());
     }
 
-    const _vmCircuitPublicInputs = enqueuedCallTrace.toVMCircuitPublicInputs(
-      constants,
-      avmExecutionEnv,
-      /*startGasLeft=*/ allocatedGas,
-      /*endGasLeft=*/ Gas.from(avmContext.machineState.gasLeft),
-      avmResult,
-    );
-
     PublicExecutor.log.verbose(
-      `[AVM] ${fnName} simulation complete. Reverted=${avmResult.reverted}. Consumed ${
+      `[AVM] ${fnName} simulation complete. Reverted=${avmCallResult.reverted}. Consumed ${
         allocatedGas.l2Gas - avmContext.machineState.gasLeft.l2Gas
       } L2 gas, ending with ${avmContext.machineState.gasLeft.l2Gas} L2 gas left.`,
     );
 
     return publicExecutionResult;
   }
+
+  /**
```

Comment:
Only used by TXE? If yes, I'd mark it as such so that we can try to remove it later.